gprofiler/profilers/profiler_base.py

```diff
@@ -207,14 +208,12 @@ def __init__(
         self,
         frequency: int,
         duration: int,
-        stop_event: Optional[Event],
-        storage_dir: str,
+        profiler_state: ProfilerState,
         insert_dso_name: bool,
-        profile_spawned_processes: bool,
         profiling_mode: str,
     ):
-        super().__init__(frequency, duration, stop_event, storage_dir, insert_dso_name, profiling_mode)
-        self._profile_spawned_processes = profile_spawned_processes
+        super().__init__(frequency, duration, profiler_state, insert_dso_name, profiling_mode)
+        self._profile_spawned_processes = profiler_state.profile_spawned_processes
```

Comment:
Don't store it - access we store the `_profiler_state` as an instance attribute and access it via that attr.